projects/distributed-tracing/src/pages/trace-detail/trace-detail.service.ts

```diff
@@ -101,6 +103,57 @@ export class TraceDetailService implements OnDestroy {
     );
   }
 
+  public fetchLogEvents(): Observable<LogEvent[]> {
+    return this.routeIds$.pipe(
+      switchMap(routeIds =>
+        this.getLogEventsGqlResponse(routeIds.traceId, routeIds.startTime).pipe(
+          map((trace: Trace) => {
+            let logEvents: LogEvent[] = [];
+            trace.spans?.forEach((span: Span) => {
+              logEvents = [
```

Comment:
why not logEvents.push ?

Reply:
yeah. or use map. We can avoid spreading the array with each iteration.

Are you trying to create a flat array?

```
const logEvents = (trace.spans ?? []).map((span => {
 return span.logEvents.results.map(( logEvent) => ({...}));
})
.flat();
```

Would this work?

Also why do we have to access `results` property directly? I believe the handler should do some initial mapping. If it does then `results` should be required.

Reply:
or this
```
```
const logEvents = (trace.spans ?? [])
.map( span => span.logEvents.results)
.flat()
.map(logEvent => ({...});
```

Reply:
Done!!